2_filtered_results/filter_data.py

```diff
@@ -0,0 +1,395 @@
+            0.4,
+            f"Z-score:{interval:.2f}={value:.3e}",
+            color="red",
+            rotation=90,
+            va="center",
+        )
+
+    ax.set_xlabel(r"RMS Forces (kcal mol$^{-1}$ $\mathrm{\AA}^{-1})$")
+    ax.set_ylabel("Count")
+    ax.set_title("Distribution of RMS Forces")
+    fig.savefig(str(output_dir / "rms_forces.png"), dpi=300, bbox_inches="tight")
+
+    # Get the data above the 95th percentile
+    cap_z_score = 4
+    df_cap_z_score = data_df[data_df["rms_forces"] > z_score_dict[cap_z_score]]
+    logger.info(f"Cutoff: {z_score_dict[cap_z_score]:.2f} kcal/(mol Angstrom)")
+    high_force_smiles: list[str] = df_cap_z_score["smiles"].tolist()
+    with open(output_dir / "high_force_smiles.json", "w") as file:
+        json.dump(high_force_smiles, file, indent=4)
+    logger.info(f"Removed {len(df_cap_z_score)} entries with high forces")
+
+    # Save a filtered dataset without the high forces
+    filtered_dataset = dataset.filter(lambda x: x["smiles"] not in high_force_smiles)
+    filtered_dataset.save_to_disk(output_dir)
+    logger.info(
+        f"Filtered dataset (containing {len(filtered_dataset)} entries) saved to {output_dir}"
+    )
+
+    # Save all of the smiles to a json file in the output directory
+    with open(output_dir / "smiles.json", "w") as file:
+        json.dump(list(filtered_dataset.unique("smiles")), file, indent=4)
+    logger.info(f"Auxiliary files saved to {output_dir}")
+
+
+def main(data_dir: pathlib.Path) -> None:
+    """Main processing function for filtering molecular datasets by forces.
+
+    Orchestrates the filtering workflow by applying Z-score-based filtering
+    to remove entries with high RMS forces from molecular datasets.
+
+    Parameters
+    ----------
+    data_dir : pathlib.Path
+        Path to the directory containing HuggingFace formatted molecular dataset.
+        The directory should contain dataset_info.json, state.json, and .arrow files.
+
+    Returns
+    -------
+    None
+
+    Notes
+    -----
+    This function performs the following workflow:
+    1. Applies Z-score filtering to remove entries with Z-score > 4 for RMS forces
+    2. Saves the filtered dataset to current working directory with name [input_dir.name]-z-score
+    3. Generates visualization of force distributions within the filtered dataset directory
+    4. Saves lists of removed and remaining SMILES within the filtered dataset directory
+
+    The percentile-based filtering function is also enabled and creates a dataset with
+    name [input_dir.name]-95thpercentile.
+
+    This function is typically called from the command-line interface but
+    can also be used programmatically when importing the module.
+
+    Examples
+    --------
+    >>> # Programmatic usage:
+    >>> data_dir = pathlib.Path("./data-raw")
+    >>> main(data_dir)
+
+    >>> # With string path:
+    >>> main(pathlib.Path("/path/to/molecular/dataset"))
+    """
+    data_dir = pathlib.Path(data_dir)
```

Comment:
If this input is already a `pathlib.Path`, I think this casting is redundant? If you want it to be `str | pathlib.Path` this is needed